### PR TITLE
Taxi game improvements

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

File: taxi/src/lib.rs [hunk 1]
```diff
@@ -9,10 +9,8 @@ use termion::event::Key;
 use termion::input::TermRead;
 use termion::raw::IntoRawMode;
 
-static ROW_SIZE: usize = 11;
-static COL_SIZE: usize = 11;
-
-#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+/// Move direction
+#[derive(PartialEq, Eq, Hash, Clone, Copy, Debug)]
```
Comment (debris) [inline taxi/src/lib.rs:13]: It's a plain c-like enum. It's better to always copy it.

File: taxi/src/lib.rs [hunk 2]
```diff
@@ -21,18 +19,20 @@ pub enum Dir {
...
+			2 => Ok(Dir::Down),
+			3 => Ok(Dir::Left),
+			_ => Err("Cannot convert u32 to `Dir`"),
 		}
 	}
 }
 
 #[derive(PartialEq, Eq, Hash, Clone, Copy)]
-pub enum Object {
+enum Object {
```
Comment (debris) [inline taxi/src/lib.rs:35]: `Object` was public but not possible to use with `Game` interface. Just unnecessary distraction

File: taxi/src/lib.rs [hunk 3]
```diff
@@ -50,26 +50,34 @@ impl fmt::Display for Object {
 	}
 }
 
-type World = Vec<Vec<Object>>;
+/// Calculates distance between two points
+pub fn distance(p0: (u32, u32), p1: (u32, u32)) -> f64 {
```
Comment (debris) [inline taxi/src/lib.rs:54]: I'm actually not sure if this should be public, cause it also feels like unnecessary distraction

File: taxi/src/lib.rs [hunk 4]
```diff
@@ -50,26 +50,34 @@ impl fmt::Display for Object {
...
-	/// Position of the player, defined as (row, column) coordinate on the world map
-	pub passenger: (u32, u32),
-	///
-	pub picked_up: bool,
-	/// Position of the player, defined as (row, column) coordinate on the world map
-	pub goal: (u32, u32),
+	position: (u32, u32),
+	/// Position of the passenger, defined as (row, column) coordinate on the world map
+	passenger: (u32, u32),
+	/// Passenger has been picked up
```
Comment (debris) [inline taxi/src/lib.rs:69]: I think that public fields also cause unnecessary distraction.

File: taxi/src/lib.rs [hunk 5]
```diff
@@ -89,24 +97,40 @@ impl Game {
...
-		) as f64).sqrt())
+		distance(self.passenger, self.position)
 	}
 
 	/// Enter move directly instead of starting an stdin loop, for instance from an automated player
 	/// Returns a bool signifying whether or not this move lead to winning the game
-	pub fn enter_move(&mut self, dir: &Dir, print: bool) -> bool {
-		self.make_move(&dir);
+	pub fn enter_move(&mut self, dir: Dir, print: bool) -> bool {
+		self.make_move(dir);
```
Comment (debris) [inline taxi/src/lib.rs:133]: it's a simple enum, so like `u8`, it's better to pass it by value, not reference ;)

File: taxi/src/lib.rs (hunk 3 above)
Comment (folsen) [reply taxi/src/lib.rs:54]: There's strategies that involve using euclidean distance as reward function, so thought I'd have it to make it simple for people, but it's not necessary to solve the game so maybe you're right.

**Pull request merged**